pkgs/development/compilers/ldc/default.nix

```diff
@@ -42,17 +47,14 @@ stdenv.mkDerivation rec {
       rm ldc-${version}-src/tests/d2/dmd-testsuite/fail_compilation/mixin_gc.d
       rm ldc-${version}-src/tests/d2/dmd-testsuite/runnable/xtest46_gc.d
       rm ldc-${version}-src/tests/d2/dmd-testsuite/runnable/testptrref_gc.d
+
+      # test depends on current year
```

Comment:
This is impure?

Reply:
So far as I can tell, the test generates documentation and compares against expected output:

https://github.com/ldc-developers/dmd-testsuite/blob/04f77cac6a01d81e5f2d44f4b479bb4eeff66503/compilable/extra-files/ddocYear-postscript.sh#L3-L6

With expected output including

https://github.com/ldc-developers/dmd-testsuite/blob/04f77cac6a01d81e5f2d44f4b479bb4eeff66503/compilable/extra-files/ddocYear.html#L540

The script seems to have support for using the token `__YEAR__` to generate the correct expectation, but the token doesn't appear in the expected output. There's a patch to update this from 2019 to 2020: https://github.com/ldc-developers/dmd-testsuite/commit/04f77cac6a01d81e5f2d44f4b479bb4eeff66503